test/DatabaseTest_Cpp.cc

```diff
@@ -230,112 +163,112 @@ TEST_CASE_METHOD(CBLTest_Cpp, "C++ Transaction With Exception", "[!throws]") {
     }
     CHECK(threw);
 
-    Document doc = db.getDocument("foo");
+    Document doc = col.getDocument("foo");
     REQUIRE(doc);
     CHECK(doc["greeting"].asString() == "Howdy!");
     CHECK(doc["meeting"] == nullptr);
 }
 
-static void createDocument(Database db, const char *docID,
+static void createDocumentDefault(Collection col, const char *docID,
                            const char *property, const char *value)
 {
     MutableDocument doc(docID);
     doc[property] = value;
-    db.saveDocument(doc);
+    col.saveDocument(doc);
 }
 
-TEST_CASE_METHOD(CBLTest_Cpp, "Legacy - C++ Database notifications") {
-    int dbListenerCalls = 0, fooListenerCalls = 0;
+TEST_CASE_METHOD(CBLTest_Cpp, "C++ Database notifications") {
```

Comment:
Same as the C ones.

A bit messy here: I think we should do this :

1. Keep the current database's listener tests until we decided to remove as there is some logic that wraps the database listener with collection listener that should be tested. Can you revert the change for this?

2. Add collection's listener tests to CollectionTest class (it seems like we don't have ones). Please check ticket for this.

Reply:
I did change the listeners such that they reflect the changes for collections. The ones in Database do use default collection now.

Reply:
On the same note as above, there might be a way to concat these using sections, checking listener for default and custom collection in one place. If is the case, I will do it as part of CBL-4194.

Reply:
It's not the same as using the default collection. There is a bit of code here that I think we should leave the original test and add the collection change listener in the Collection Test class:

https://github.com/couchbase/couchbase-lite-C/blob/master/src/CBLDatabase_CAPI.cc#L413-L434